### PR TITLE
Batch serving and AWS deployment

### PR DESCRIPTION
**Pull request by adiso75**
@jillnogold  please review the batch serving first. 
@Tankilevitch  for the AWS doc, we need to wait to the community edition and then we can release it to the docs

### CODE REVIEW

File: docs/serving/batch-serving.md [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+```python
+auto_trainer = mlrun.import_function("hub://auto_trainer")
+```
+
+### Input dataset
+
+In this example we're using a csv file (stored on wassabi object store) as an input dataset for the prediction <br>
+
+```python
+DATASET_URL = 'https://s3.wasabisys.com/iguazio/data/function-marketplace-data/xgb_trainer/classifier-data.csv'
```
Comment (gilad-shaham) [inline docs/serving/batch-serving.md:62]: Should use MLRun function to get sample data and not the full URL

File: docs/serving/batch-serving.md [hunk 2]
```diff
@@ -0,0 +1,611 @@
...
+## Overview
+
+Batch inference or offline inference addresses the need to run machine learning model on large datasets. <br> 
+
+It is the process of generating predictions on a batch of observations. The batch jobs are typically generated on some recurring schedule (e.g. hourly, daily). <br>
+These predictions are then stored in a database or a file and can be made available to developers or end users. Batch inference may sometimes take advantage <br>
+of big data technologies such as Spark to generate predictions. This allows data scientists and machine learning engineers to take advantage of scalable compute <br>
+resources to generate many predictions at once. As opposed to real time inference real-time inference where the goal is usually to optimize the number of transactions per second <br>
+that the model can process,  With batch inference, the goal is usually tied to time constraints and the service-level agreement (SLA) for the job.  <br>
+in addition, the output of batch inference goes to a file or a table in a database vs sending a response back to an online application
```
Comment (gilad-shaham) [inline docs/serving/batch-serving.md:14]: Why so many line breaks?

File: docs/serving/batch-serving.md [hunk 3]
```diff
@@ -0,0 +1,611 @@
...
+function closePanel(el) {
+  const panelName = "#" + el.getAttribute('paneName')
+  const resultPane = document.querySelector(panelName + "-pane");
+  if (!resultPane.classList.contains("hidden")) {
+    resultPane.classList.add("hidden");
+  }
+}
+
+</script>
+<div class="master-wrapper">
```
Comment (gilad-shaham) [inline docs/serving/batch-serving.md:421]: All of this renders in the markdown as text, need to fix

File: docs/serving/batch-serving.md [hunk 4]
```diff
@@ -0,0 +1,611 @@
+# Batch inference 
+
+This topic describes how to run a batch predict job along with drift analysis
```
Comment (gilad-shaham) [inline docs/serving/batch-serving.md:3]: Not sure if we need this sentence

File: docs/serving/batch-serving.md [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+
+## Overview
+
+Batch inference or offline inference addresses the need to run machine learning model on large datasets. <br> 
+
+It is the process of generating predictions on a batch of observations. The batch jobs are typically generated on some recurring schedule (e.g. hourly, daily). <br>
```
Comment (gilad-shaham) [inline docs/serving/batch-serving.md:9]: Should mention early on that this is done by creating a job

File: docs/serving/batch-serving.md [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+
+
+
+## Run prediction
+
+you can either use a trained model that is already registered in MLRun or just pass the model URI
+
+
+```python
+model_path = 'store://artifacts/auto-trainer-admin/MyModel:cf9538d81a2a4ce9a2035ad3d8b90456'
```
Comment (gilad-shaham) [inline docs/serving/batch-serving.md:251]: Should reference a model from the samples and not something that is not runnable if I take the code as-is

File: docs/serving/batch-serving.md [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+> 2022-08-11 06:01:09,456 [info] task scheduled, {'schedule': '*/30 * * * *', 'project': 'batch-inference-admin', 'name': 'auto-trainer-predict'}
+
+
+## Viewing the drift result
+
+By default, the auto_trainer functions analyze drift by comparing the input and the sample dataset statistics. <br> 
+it triggers a drift analysis job which generates 3 output files that can be viewed under the artifacts tab for the prediction job
+* drift_table_plot - shows expected vs actual statistics for every feature as a plot.
+* drift_results - shows the drift result per feature as a JSON format so you can use it to process this data and apply your logic on top of it.
+* dataset_statistics - shows the statistics of the input dataset.
```
Comment (gilad-shaham) [inline docs/serving/batch-serving.md:607]: Should also show how to present the artifact here

**Review by gilad-shaham — CHANGES_REQUESTED** [review]

File: docs/serving/batch-serving.md [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+## Auto-traininer function for prediction
+
+The [Auto-trainer](https://www.mlrun.org/marketplace/functions/master/auto_trainer/latest/example/) is a function in the mlrun marketplace that is used for both training and prediction. <br>
+This function saves you the need to write your own predict function and in addition it also runs a drift analysis by default. <br>
+The drift compares the prediction data with the sample/test dataset <br>
+In order to run batch predict one should use the predict handler <br>
+
+The function supports the following frameworks:
+* XGBoost
+* sklearn
```
Comment (guy1992l) [inline docs/serving/batch-serving.md:25]:
```suggestion
* SciKit-Learn
```

File: docs/serving/batch-serving.md [hunk 9]
```diff
@@ -0,0 +1,611 @@
...
+These predictions are then stored in a database or a file and can be made available to developers or end users. Batch inference may sometimes take advantage <br>
+of big data technologies such as Spark to generate predictions. This allows data scientists and machine learning engineers to take advantage of scalable compute <br>
+resources to generate many predictions at once. As opposed to real time inference real-time inference where the goal is usually to optimize the number of transactions per second <br>
+that the model can process,  With batch inference, the goal is usually tied to time constraints and the service-level agreement (SLA) for the job.  <br>
+in addition, the output of batch inference goes to a file or a table in a database vs sending a response back to an online application
+
+## Auto-traininer function for prediction
+
+The [Auto-trainer](https://www.mlrun.org/marketplace/functions/master/auto_trainer/latest/example/) is a function in the mlrun marketplace that is used for both training and prediction. <br>
```
Comment (guy1992l) [inline docs/serving/batch-serving.md:18]:
```suggestion
The [Auto-trainer](https://www.mlrun.org/marketplace/functions/master/auto_trainer/latest/example/) is a function in the mlrun marketplace that is used for multiple steps in the model life-cycle including training, evaluation and prediction. <br>
```

File: docs/serving/batch-serving.md [hunk 10]
```diff
@@ -0,0 +1,611 @@
...
+This topic describes how to run a batch predict job along with drift analysis
+
+## Overview
+
+Batch inference or offline inference addresses the need to run machine learning model on large datasets. <br> 
+
+It is the process of generating predictions on a batch of observations. The batch jobs are typically generated on some recurring schedule (e.g. hourly, daily). <br>
+These predictions are then stored in a database or a file and can be made available to developers or end users. Batch inference may sometimes take advantage <br>
+of big data technologies such as Spark to generate predictions. This allows data scientists and machine learning engineers to take advantage of scalable compute <br>
+resources to generate many predictions at once. As opposed to real time inference real-time inference where the goal is usually to optimize the number of transactions per second <br>
```
Comment (guy1992l) [inline docs/serving/batch-serving.md:12]:
```suggestion
resources to generate many predictions at once. As opposed to real-time inference where the goal is usually to optimize the number of transactions per second <br>
```

File: docs/serving/batch-serving.md [hunk 11]
```diff
@@ -0,0 +1,611 @@
...
+that the model can process,  With batch inference, the goal is usually tied to time constraints and the service-level agreement (SLA) for the job.  <br>
```
Comment (guy1992l) [inline docs/serving/batch-serving.md:13]:
```suggestion
that the model can process, with batch inference, the goal is usually tied to time constraints and the service-level agreement (SLA) for the job.  <br>
```

File: docs/serving/batch-serving.md [hunk 12]
```diff
@@ -0,0 +1,611 @@
...
+in addition, the output of batch inference goes to a file or a table in a database vs sending a response back to an online application
+
+## Auto-traininer function for prediction
+
+The [Auto-trainer](https://www.mlrun.org/marketplace/functions/master/auto_trainer/latest/example/) is a function in the mlrun marketplace that is used for both training and prediction. <br>
+This function saves you the need to write your own predict function and in addition it also runs a drift analysis by default. <br>
+The drift compares the prediction data with the sample/test dataset <br>
```
Comment (guy1992l) [inline docs/serving/batch-serving.md:20]:
```suggestion
The drift compares the prediction data with the sample/training dataset <br>
```

File: docs/serving/batch-serving.md (hunk 7 above)
Comment (guy1992l) [reply docs/serving/batch-serving.md:607]: should be done with something like:

```
predict_run_object.artifact('drift_table_plot').show()
```

In the next section Drift table plot example

**Review by guy1992l — COMMENTED** [review]
minor comments, looks good overall :)

File: docs/install/aws-deployment.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
+# Install MLRun on AWS
+For AWS users, the easiest way to install MLRun is to use a native AWS deployment. This option deploys MLRun on an AWS EKS service using a cloud formation stack.
+
+## Prerequisites
+- AWS account <br>
+  See [how to create a new AWS account](https://aws.amazon.com/premiumsupport/knowledge-center/create-and-activate-aws-account/).
```
Comment (gilad-shaham) [inline docs/install/aws-deployment.md:6]: AWS Account with administrative privileges to create an EKS cluster

File: docs/install/aws-deployment.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
```
Comment (gilad-shaham) [inline docs/install/aws-deployment.md:7]: Also, does the user need to have docker hub account or do we use ECR by default?

File: docs/install/aws-deployment.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+> **The MLRun software is free of charge, however, there is a cost for the AWS hardware and the EKS service.**
+
+
+## Click on the AWS icon to deploy Iguazio
+<a href="https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/quickcreate?templateUrl=https%3A%2F%2Fmlrun-kit-alexp.s3.us-east-2.amazonaws.com%2Fquickstart-amazon-eks%2Ftemplates%2Figuazio-mlrun-kit-entrypoint-new-vpc.template.yaml&stackName=MLrun-community%20&param_AdditionalEKSAdminUserArn=&param_AvailabilityZones%5B%5D=&param_ClusterDomain=&param_DeployMLRunKit=true&param_EKSClusterName=&param_KeyPairName=&param_MLrunKitVersion=&param_NodeInstanceFamily=Standard&param_NodeInstanceType=m5.2xlarge&param_NumberOfAZs=3&param_NumberOfNodes=3&param_ProvisionBastionHost=Disabled&param_RegistryDomainName=index.docker.io&param_RegistryEmail=&param_RegistrySuffix=%2Fv1%2F&param_RegistryUsername=&param_RemoteAccessCIDR="><img src="../_static/images/aws-icon.png"></img></a>
```
Comment (gilad-shaham) [inline docs/install/aws-deployment.md:13]: Image should change to something nicer like this
Launch Stack

File: docs/install/aws-deployment.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+* Real time serverless framework  (Nuclio)
+* Spark operator
+* Jupyter lab
+
+## Get started
+Once the deployment is done go to the output tab for the satck you've created. You'll see the links for the MLRun UI, Jupyter and the Kubeconfig command <br>
+We encourgae you to go through the quick start and the other tutorials as shown in the documentaion. note that those tutorials and demos comes built-in with Jupyter under the root folder of Jupyter.
+
+
+## AWS services to be aware of
```
Comment (gilad-shaham) [inline docs/install/aws-deployment.md:34]: Should add an explicit section called "Cleaning" and add instructions where to delete the stack

**Review by gilad-shaham — CHANGES_REQUESTED** [review]

File: docs/install/aws-deployment.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+
+
+## Click on the AWS icon to deploy Iguazio
+<a href="https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/quickcreate?templateUrl=https%3A%2F%2Fmlrun-kit-alexp.s3.us-east-2.amazonaws.com%2Fquickstart-amazon-eks%2Ftemplates%2Figuazio-mlrun-kit-entrypoint-new-vpc.template.yaml&stackName=MLrun-community%20&param_AdditionalEKSAdminUserArn=&param_AvailabilityZones%5B%5D=&param_ClusterDomain=&param_DeployMLRunKit=true&param_EKSClusterName=&param_KeyPairName=&param_MLrunKitVersion=&param_NodeInstanceFamily=Standard&param_NodeInstanceType=m5.2xlarge&param_NumberOfAZs=3&param_NumberOfNodes=3&param_ProvisionBastionHost=Disabled&param_RegistryDomainName=index.docker.io&param_RegistryEmail=&param_RegistrySuffix=%2Fv1%2F&param_RegistryUsername=&param_RemoteAccessCIDR="><img src="../_static/images/aws-icon.png"></img></a>
+
+
+After clicking the icon, the browser opens your AWS login page and directs you to the CloudFormation stack page. <br>
+Fill in the form and click **Done**. MLRun creates a VPC with an EKS cluster and deploys all the services on top of it. <br>
+
+The key components that being deployed on your EKS cluster are:
```
Comment (jillnogold) [inline docs/install/aws-deployment.md:19]: The key components deployed on your EKS cluster are:

File: docs/install/aws-deployment.md [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+
+* EKS 
+* MLRun server (including the feature store and the MLrun graph)
```
Comment (jillnogold) [inline docs/install/aws-deployment.md:22]: MLRun

File: docs/install/aws-deployment.md [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+* MLRun UI
+* Kubeflow pipeline
+* Real time serverless framework  (Nuclio)
+* Spark operator
+* Jupyter lab
+
+## Get started
+Once the deployment is done go to the output tab for the satck you've created. You'll see the links for the MLRun UI, Jupyter and the Kubeconfig command <br>
```
Comment (jillnogold) [inline docs/install/aws-deployment.md:30]: Go to the output tab for the stack you created. There are links for the MLRun UI, Jupyter and the Kubeconfig command. <br>
It's recommended to go through the quick start and the other tutorials as shown in the documentation. These tutorials and demos comes built-in with Jupyter under the root folder of Jupyter.

File: docs/install/aws-deployment.md [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+## AWS services to be aware of
+When installing the MLRun Community Edition via Cloud Formation, several storage resources are created, some of which persist even after uninstalling the stack:
+* PVs via AWS storage provider: Used to hold the file system of the stacks pods, including the MySQL database of MLRun. These are deleted once the stack is uninstalled.
+* S3 Bucket: A bucket named mlrun is created in the AWS account that installs the stack. The bucket is used for MLRun’s artifact storage, and is not deleted when uninstalling the stack. The user must empty the bucket and delete it.
+* Container Images in ECR: When building and deploying MLRun and Nuclio functions via the MLRun Community Edition, the function images are stored in an ECR belonging to the AWS account that installs the stack. These images persist in the account’s ECR and are not deleted either.
+
+## How to configure the online feature store
+
+The feature store can store data on a fast key value database table for quick serving. This online feature store capability requires an external key value database. <br>
+Currently the MLrun feature store supports Iguazios' key value database.<br>
```
Comment (jillnogold) [inline docs/install/aws-deployment.md:43]: Currently the MLrun feature store supports Iguazios' key value database. The Iguazio roadmap includes support for the Redis database very soon, as well as other key value databases down the road. <br>
For more information on the feature store, see [Feature store](https://docs.mlrun.org/en/latest/feature-store/feature-store.html)

**Pull request closed without merging**